app/connectors/connector-verifier/src/main/java/io/syndesis/verifier/api/ComponentMetaDataAdapter.java

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright (C) 2016 Red Hat, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.syndesis.verifier.api;
+
+import java.util.Map;
+
+import io.syndesis.core.CollectionsUtils;
+import org.apache.camel.CamelContext;
+import org.apache.camel.Component;
+import org.apache.camel.component.extension.MetaDataExtension;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public abstract class ComponentMetaDataAdapter implements MetadataAdapter {
+    private static final Logger LOG = LoggerFactory.getLogger(ComponentMetaDataAdapter.class);
+
+
+    private final Class<? extends MetaDataExtension> metaDataExtensionClass;
+
+    @SuppressWarnings("PMD.AvoidUsingVolatile")
+    private volatile MetaDataExtension metaDataExtension;
```

Comment:
I wonder if [memoization](https://en.wikipedia.org/wiki/Memoization) would be more suited to this use case, perhaps a simple one using `ConcurrentHashMap::computeIfAbsent`